src/createReplaceTokens.js

```diff
@@ -27,13 +29,16 @@ module.exports = function(isVar, getVar, undefinedVarsReturnEmpty) {
   var replaceTokensInObject;
   var replaceTokensInArray;
   var replaceTokens;
+  var variablesBeingRetrieved = [];
 
   var getVarValue = function(token, variableName, syntheticEvent) {
     if (!isVar(variableName)) {
       return token;
     }
 
+    variablesBeingRetrieved.push(variableName);
     var val = getVar(variableName, syntheticEvent);
+    variablesBeingRetrieved.pop();
```

Comment:
Maintaining this array is a bit of a cost, but probably worth having a useful error message in the case of a circular dependency.